CenterForOpenScience/osf.io
# [ENG-415][ENG-798] Add periodic task that deactivates requested accounts with tests

## Purpose

To speed up the deactivation request process we want to automatically delete users. Now users who request deactivation get emailed or deactivated at midnight each night.

## Changes

- add periodic script and tests
- remove now unnecessary emails sending and throttling tests

## QA Notes

To test this issue you need to have an admin user with permissions that can deactivate an account. Once you do that you can create some test users, make sure some are admin on nodes, both the only admin and shared with other admins. Have though users request deletion and wait until midnight for the period task to run. The task should send our admin emails and deactivate completely those without nodes, preprints, groups etc.

## Documentation

Not user facing, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-415
